src/components/Warehouse/AddWarehouse.js

```diff
@@ -0,0 +1,302 @@
+import { useState } from "react";
+import "./AddWarehouse.scss";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+import BackIcon from "../../assets/icons/arrow_back-24px.svg";
+import ErrorIcon from "../../assets/icons/error-24px.svg";
+import Footer from "../Footer/Footer";
+import Header from "../Header/Header";
+
+const AddWarehouse = () => {
+  const navigate = useNavigate();
+
+  const initialState = {
+    warehouse_name: "",
+    address: "",
+    city: "",
+    country: "",
+    contact_name: "",
+    contact_position: "",
+    contact_phone: "",
+    contact_email: "",
+  };
+
+  // State to keep track of form data
+  const [formData, setFormData] = useState(initialState);
+
+  //state to keep track of user errors
+  const [errors, setErrors] = useState({});
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData((prevFormData) => ({ ...prevFormData, [name]: value }));
+  };
+
+  //Function to Validate user inputs
+  const validateForm = (inputValues) => {
+    const errors = {};
+    const validations = {
+      warehouse_name: "Warehouse name missing",
+      address: "Warehouse address missing",
+      city: "Warehouse city missing",
+      country: "Warehouse country missing",
+      contact_name: "Contact name missing",
+      contact_position: "Contact position missing",
+      contact_phone: "Contact phone missing",
+      contact_email: "Contact email missing",
+    };
+
+    // Email validation regex pattern
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+    // Phone number validation regex pattern
+    const phoneRegex = /^\d{10}$/;
+
+    // Check for missing required fields
+    Object.keys(validations).forEach((key) => {
+      if (!inputValues[key]) {
+        errors[key] = validations[key];
+      }
+    });
+
+    // Validate email format only if the email field is not empty
+    if (
+      inputValues.contact_email &&
+      !emailRegex.test(inputValues.contact_email)
+    ) {
+      errors.contact_email = "Please enter a valid email address";
+    }
+
+    // Validate phone number format only if the phone field is not empty
+    if (
+      inputValues.contact_phone &&
+      !phoneRegex.test(inputValues.contact_phone)
+    ) {
+      errors.contact_phone = "Please enter a valid 10-digit phone number";
+    }
+
+    console.log(errors);
+    return errors;
+  };
+
+  //Function to handle form submission
+  const handleSubmit = async (e) => {
+    try {
+      e.preventDefault();
+      setErrors(validateForm(formData));
+      await axios.post("http://localhost:8080/api/warehouses", formData);
```

Comment:
You should be validating if errors exist before sending a POST request. 

I was able to submit an empty form. Even though the network request correctly returned 400 Bad Request, we should not even be sending an API call if validations dont pass. We save unnecessary network requests in the process